web/packages/teleport/src/services/locks/types.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright 2023 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// LockKind is the expected backend value
+// to define the type of lock to be locked.
+export type LockKind =
+  | 'user'
+  | 'role'
+  | 'login'
+  | 'node'
+  | 'mfa_device'
+  | 'windows_desktop'
+  | 'access_request'
+  | 'device'; // trusted devices
+
+export type Lock = {
+  name: string;
+  message: string;
+  expires: string;
+  createdAt: string;
+  createdBy: string;
+  targetLookup: Partial<Record<LockKind, string>>;
+  targets: LockTarget[];
+};
+
+export type LockTarget = {
+  kind: LockKind;
+  name: string;
```

Comment:
This bit us in the past, right? It's not always a name.

Also, I'm fine with this as-is for simplicity, but note that the data model for the frontend doesn't match the backend now. On the backend, a lock target can have multiple different resources (so long as they are different types).

Reply:
>This bit us in the past, right? It's not always a name.

i think this only applies to mapping backend with frontend fields, which isn't the case here. this type is only for frontend use. in the api response we get a mapping of targets, it wasn't very useful in its map form so I converted it into a list of LockTarget and added it as an extra field to the frontend model. i did it in this layer so we don't have to always process it when we use it in components.

we have a lot of frontend models that don't match the backend b/c we customize it further to suit our needs. i thought it was okay to do but maybe it shouldn't be done in the service layer. @ryanclark thoughts?